cdap-support-bundle/src/main/java/io/cdap/cdap/app/guice/SupportBundleRuntimeServiceModule.java

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright © 2021 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.app.guice;
+
+import com.google.inject.AbstractModule;
+import com.google.inject.Module;
+import com.google.inject.PrivateModule;
+import com.google.inject.Provides;
+import com.google.inject.Scopes;
+import com.google.inject.multibindings.Multibinder;
+import com.google.inject.name.Named;
+import com.google.inject.name.Names;
+import com.google.inject.util.Modules;
+import io.cdap.cdap.app.store.Store;
+import io.cdap.cdap.common.conf.CConfiguration;
+import io.cdap.cdap.common.conf.Constants;
+import io.cdap.cdap.common.conf.Constants.SupportBundle;
+import io.cdap.cdap.common.runtime.RuntimeModule;
+import io.cdap.cdap.common.utils.Networks;
+import io.cdap.cdap.gateway.handlers.CommonHandlers;
+import io.cdap.cdap.internal.app.runtime.artifact.ArtifactRepository;
+import io.cdap.cdap.internal.app.runtime.artifact.ArtifactRepositoryReader;
+import io.cdap.cdap.internal.app.runtime.artifact.AuthorizationArtifactRepository;
+import io.cdap.cdap.internal.app.runtime.artifact.DefaultArtifactRepository;
+import io.cdap.cdap.internal.app.runtime.artifact.LocalArtifactRepositoryReader;
+import io.cdap.cdap.internal.app.services.SupportBundleInternalService;
+import io.cdap.cdap.internal.app.store.DefaultStore;
+import io.cdap.cdap.logging.read.FileLogReader;
+import io.cdap.cdap.logging.read.LogReader;
+import io.cdap.cdap.security.impersonation.DefaultOwnerAdmin;
+import io.cdap.cdap.security.impersonation.DefaultUGIProvider;
+import io.cdap.cdap.security.impersonation.OwnerAdmin;
+import io.cdap.cdap.security.impersonation.UGIProvider;
+import io.cdap.cdap.support.handlers.SupportBundleHttpHandler;
+import io.cdap.cdap.support.task.factory.SupportBundlePipelineInfoTaskFactory;
+import io.cdap.cdap.support.task.factory.SupportBundleSystemLogTaskFactory;
+import io.cdap.cdap.support.task.factory.SupportBundleTaskFactory;
+import io.cdap.http.HttpHandler;
+
+import java.net.InetAddress;
+import java.net.InetSocketAddress;
+
+/**
+ * SupportBundle Service Guice Module.
+ */
+public final class SupportBundleRuntimeServiceModule extends RuntimeModule {
```

Comment:
This list looks strange. Usually distributed modules install same interfaces but different implementation